### PR TITLE
PostgreSQL 12 compat, mark II

### PR DESCRIPTION
**Pull request by dleavitt**
Exactly the same as #22 except I've rebased it on top of master. Some of the tests continue to sporadically fail, I don't think that's a new thing though?

Also looks like I made a commit redundant with something on master, which had the net effect of changing the quote type. Oh well! Let's get this fella merged and released!

### CODE REVIEW

**Comment by betesh** [comment]
* Instead of removing sprockets, put a file in the location that sprockets is complaining about.  It can probably be empty.
* I'm still confused about what's broken here.  Can you push a commit with a test that fails on pg 12 but passes on older versions, in a separate PR so that we can all see the results from travis?

**Comment by dleavitt** [comment]
what's failing is just rubocop from me messing up your quotes. I'll just rebase out that commit, it doesn't do anything (sprockets was already removed on your master.)

**Comment by dleavitt** [comment]
donzo. things are looking greener now.

**Comment by betesh** [comment]
OK but the trouble is that I still don't see a test that passed with older pg versions, fails on pg 12, and then passes on pg 12 with this fix.  So I don't know whether by fixing something for pg 12, we're breaking older versions.  Can you push a commit with such a test?

**Comment by dleavitt** [comment]
@betesh I'm guessing the travis setup doesn't actually test against multiple versions of PG? Looking into it now.

**Comment by dleavitt** [comment]
Confirmed:

Screenshot 2020-03-18 17 32 42

Seems like [the workaround](https://github.com/travis-ci/travis-ci/issues/954) isn't effective (anymore?)

In any case, this branch passes CI on 9.6 and passes against 12.2 on my local machine (master fails) so seems safe to merge?

**Comment by betesh** [comment]
Looks like travis doesn't support any pg versions after 9.6.  See https://travis-ci.org/github/on-site/active_record-postgres-constraints/builds/664206787 (built for https://github.com/on-site/active_record-postgres-constraints/compare/b5332920919db2e1523c61c31759744391c5993f...fix-testing-with-multiple-versions-of-postgres)

**Comment by betesh** [comment]
I'm going to give this a try with the approach documented at https://travis-ci.community/t/install-postgresql-11/3894.

**Comment by betesh** [comment]
@dleavitt Please rebase this onto the fix-testing-with-multiple-versions-of-postgres branch, where builds are green except for pg 12.  If build is green after rebasing then we can merge this.

> Some of the tests continue to sporadically fail, I don't think that's a new thing though?

Correct, we've had intermittent failures for a while, unfortunately.

**Comment by dleavitt** [comment]
A bunch of tests are still failing (though some of the PG12 ones are passing! progress?) Not sure if these are the spurious ones or not.

Probably unrelated, but I don't think the `BUNDLE_DISABLE_EXEC_LOAD=true` env var was set up correctly, so I also tried fixing that.

**Comment by dleavitt** [comment]
Can we get this merged? I don't think it causes any regressions, and it adds support for PG12

**Comment by betesh** [comment]
Looks like tests didn't trigger for the last push on travis.  See if you can trigger them and once it's green, I'll merge it.

**Comment by betesh** [comment]
It's building now.  The travis integration had broken, something to do with travis-ci.org vs. travis-ci.com.  It's fixed now.

**Comment by betesh** [comment]
I'm getting a release ready...

**Comment by betesh** [comment]
Released [0.2.1](https://rubygems.org/gems/active_record-postgres-constraints/versions/0.2.1)

File: .travis.yml [hunk 1]
```diff
@@ -24,7 +22,18 @@ matrix:
   - rvm: 2.4.0
     gemfile: gemfiles/rails_6.0.gemfile
 before_install:
-  - sudo service postgresql stop && sudo service postgresql start $POSTGRESQL_VERSION
+  - wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | sudo apt-key add -
+  - echo "deb http://apt.postgresql.org/pub/repos/apt/ `lsb_release -cs`-pgdg main" | sudo tee /etc/apt/sources.list.d/pgdg.list
+  - sudo apt update
+  - sudo apt install -yq --no-install-suggests --no-install-recommends postgresql-${POSTGRESQL_VERSION} postgresql-client-${INSTALL_VERSION}
```
Comment (adis-io) [inline .travis.yml:31]: @betesh 
`INSTALL_VERSION`
Is this typo?

Comment (betesh) [reply .travis.yml:31]: Yes.  Should be POSTGRESQL_VERSION

Comment (betesh) [reply .travis.yml:31]: Fixed in #33

**Pull request merged**